src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -0,0 +1,424 @@
+                    throw CreateSslException("SSL Handshake failed", error);                   
+                }
+            }
+        
+            return ((libssl.SSL_state(context.sslPtr) == libssl.SslState.SSL_ST_OK)) ? true : false;
+
+        }
+
+        internal static int Encrypt(IntPtr handlePtr, IntPtr buffer, int count, int bufferCapacity)
+        {
+            SslContext context = Marshal.PtrToStructure<SslContext>(handlePtr);  
+
+            var retVal = libssl.SSL_write(context.sslPtr, buffer, count);
+
+            if (retVal != count)
+            {
+                throw CreateSslException("OpenSsl::Encrypt failed");
+            }
+
+            int capacityNeeded = libssl.BIO_ctrl_pending(context.writeBioPtr);      
+
+            if (retVal == count)
+            {
+                if (capacityNeeded > bufferCapacity)
+                {
+                    throw CreateSslException("OpenSsl::Encrypt outBufferPtr should be null");
+                }
+
+                IntPtr outBufferPtr = buffer;
+
+                retVal = BioRead(context.writeBioPtr, outBufferPtr, capacityNeeded);
+
+                if (retVal < 0)
+                {
+                    throw CreateSslException("OpenSsl::Encrypt failed");
+                }
+            }
+
+            return retVal;
+        }
+
+        internal static int Decrypt(IntPtr sslContextPtr, IntPtr outBufferPtr, int count)
+        {
+            SslContext context = Marshal.PtrToStructure<SslContext>(sslContextPtr);
+
+            var retVal = BioWrite(context.readBioPtr, outBufferPtr, count);
+
+            if (retVal == count)
+            {
+                retVal = libssl.SSL_read(context.sslPtr, outBufferPtr, retVal);
+
+                if (retVal > 0) count = retVal;
+            }
+
+            if (retVal != count)
+            {
+                throw CreateSslException("OpenSsl::Decrypt failed");
+            }
+            return retVal;
+        }
+
+        internal static IntPtr GetPeerCertificate(IntPtr sslContextPtr)
+        {
+            SslContext context = Marshal.PtrToStructure<SslContext>(sslContextPtr);
+            IntPtr sslPtr = context.sslPtr;
+            IntPtr certPtr = libssl.SSL_get_peer_certificate(sslPtr);
+            return certPtr;
+        }
+
+        internal static libssl.SSL_CIPHER GetConnectionInfo(IntPtr sslContextPtr)
+        {
+            SslContext context = Marshal.PtrToStructure<SslContext>(sslContextPtr);
+            IntPtr sslPtr = context.sslPtr;
+            IntPtr cipherPtr = libssl.SSL_get_current_cipher(sslPtr);
+            var cipher = new libssl.SSL_CIPHER();
+            if (IntPtr.Zero != cipherPtr)
+            {
+                cipher = Marshal.PtrToStructure<libssl.SSL_CIPHER>(cipherPtr);
+            }
+
+            return cipher;
+        }
+
+        internal static byte[] GetCertificateRawData(IntPtr certPtr)
```

Comment:
This already exists in the current crypto interop.  It's also a bit unclear why you'd need it... an X509Certificate2 can be made directly from `certPtr`.